crates/sui-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -319,11 +319,15 @@ impl SharedObjectCongestionTracker {
         execution_time_estimator: Option<&ExecutionTimeEstimator>,
         cert: &VerifiedExecutableTransaction,
     ) {
+        let shared_input_objects: Vec<_> = cert.shared_input_objects().collect();
+        if shared_input_objects.is_empty() {
```

Comment:
I guess this is the canonical way if someone really needs to check if a txn contains shared input objects?

Reply:
Yes. I expect that in all cases where you need to check this, you would also want to use them if they are not empty.